tests/local_fork_test.go

```diff
@@ -0,0 +1,149 @@
+	"strings"
+	"go.cryptoscope.co/luigi"
+	"go.cryptoscope.co/margaret"
+	"go.cryptoscope.co/ssb/internal/storedrefs"
+	"go.cryptoscope.co/ssb/internal/mutil"
+	"go.cryptoscope.co/ssb/sbot"
+	"context"
+	"github.com/stretchr/testify/assert"
+	refs "go.mindeco.de/ssb-refs"
+)
+
+func TestStartup (t *testing.T) {
+	a := assert.New(t)
+	var err error
+	session := newSession(t, nil, nil)
+	session.startGoBot()
+	bot := session.gobot
+	feedID := bot.KeyPair.ID()
+	a.NotNil(bot)
+	botlog, err := getFeed(bot, feedID)
+	a.NoError(err)
+	a.NotNil(botlog)
+	a.EqualValues(-1, botlog.Seq(), "maggie seqno of fresh log should be -1")
+	// print empty log, note seqno
+	printLog(bot, feedID)
+
+	// note (2022-02-14): this is maybe not the same mechanism of publishing as the route used when running via muxrpcs?
+	// post a message
+	_, err = bot.PublishLog.Append(refs.Post{Type: "post", Text: "1 hello world!"})
+	a.NoError(err)
+	a.EqualValues(0, botlog.Seq(), "maggie seqno of log with 1 message should be 0")
+	_, err = bot.PublishLog.Append(refs.Post{Type: "post", Text: "2 hello world!"})
+	a.NoError(err)
+	a.EqualValues(1, botlog.Seq(), "maggie seqno of log with 2 messages should be 1")
+	fmt.Println(feedID.String())
+	// print the latest seqno
+	printLog(bot, feedID)
+	// close the go bot
+	bot.Shutdown()
+	session.wait()
+	fmt.Println(strings.Repeat("=", 15))
+	fmt.Println(strings.Repeat("=", 15))
+	fmt.Printf("%sRESTART%s\n", strings.Repeat("=", 5), strings.Repeat("=", 3))
+	fmt.Println(strings.Repeat("=", 15))
+	fmt.Println(strings.Repeat("=", 15))
+	// start the go bot again
+	session.startGoBot()
+	bot = session.gobot
+	botlog, err = getFeed(bot, feedID)
+	a.NoError(err)
+	a.NotNil(botlog)
+	a.EqualValues(1, botlog.Seq(), "maggie seqno of log with 2 messages should be 1")
+	// print the latest seqno
+	printLog(bot, feedID)
+	// post another message
+	_, err = bot.PublishLog.Append(refs.Post{Type: "post", Text: "3 hello world!"})
+	a.NoError(err)
+	a.EqualValues(2, botlog.Seq(), "maggie seqno of log with 3 messages should be 2")
+	fmt.Println(feedID.String())
+	// print the latest seqno
+	printLog(bot, feedID)
+	// post another message
+	_, err = bot.PublishLog.Append(refs.Post{Type: "post", Text: "4 hello world!"})
+	a.NoError(err)
+	a.EqualValues(3, botlog.Seq(), "maggie seqno of log with 4 messages should be 3")
+	// print the latest seqno
+	printLog(bot, feedID)
+	bot.Shutdown()
+}
+
+// error wrap - a helper util
+// string header will be prefixed before each message. typically it is the context we're generating errors within.
+// msg is the specific message, err is the error (if passed)
+func ew(header string) func(msg string, err ...error) error {
+    return func(msg string, err ...error) error {
+        if len(err) > 0 {
+            return fmt.Errorf("[gossb: %s] %s (%w)", header, msg, err[0])
+        }
+        return fmt.Errorf("[gossb: %s] %s", header, msg)
+    }
+}
+
+func printLog(bot *sbot.Sbot, feedid refs.FeedRef) error {
```

Comment:
is this a copy of this?

https://github.com/cryptoscope/ssb/blob/a846f1659c8726518aec8c081b8a980cc79c35ac/internal/testutils/streamMessageLog.go#L19

Reply:
oh! no, this is something i concocted during ngi, copied from the [treasure map](https://dev.scuttlebutt.nz/#/golang/?id=debug-print-an-entire-log-using-its-feedref)! i'll remove it now though, it's not needed anymore because i now know what's going on lol